src/igmpproxy.c

```diff
@@ -289,17 +289,23 @@ void igmpProxyRun(void) {
             }
         }
 
-        /* aimwang: call rebuildIfVc */
-        if (config->rescanVif)
-            rebuildIfVc();
-
-        // Prepare timeout...
-        secs = timer_nextTimer();
-        if(secs == -1) {
-            timeout = NULL;
+        if (config->rescanVif) (void)0;
```

Comment:
This does nothing.

Reply:
not yet, i put it in there to prevent compiler warning about unused variable config (which I don't want to remove, as it really should be in there).. It will be used to initialize timer to rescan interfaces in system.

Reply:
Also, the option rescanvif is still being loaded by loadConfig(). So if somebody actually enables it, having the old if (config->rescanVif) rebuildIfVc() will cause bad things as it would in the current version, now it won't.

In newer version it will become:
// Set rescanvif or rescanconf timer.
        if (!config->rescanConf && config->rescanVif > 0 && timer_leftTimer(rescanvif_timer) == -1) {
            rescanvif_timer=timer_setTimer(config->rescanVif, (timer_f)rebuildIfVc, NULL);
        }
        if (config->rescanConf > 0 && timer_leftTimer(rescanconf_timer) == -1) {
            rescanconf_timer=timer_setTimer(config->rescanConf, (timer_f)reloadConfig, NULL);
        }

Reply:
So is `rescanvif` option with this change still working as before?

Reply:
Well yes and no. Currently it is not working correctly. There are several issues. First it is not building a new list of interfaces to compare with old and acti accordingly. It is only making interfaces hidden, not going through the process of removing routes, sending leaves etc for upstream interfaces. It is not doing reevalution of groups joined when downstream interface transitions to disabled. The only thing currently done is:
leaveMcGroup( getMcGroupSock(), Dp, allrouters_group ); delvif
Second when interface are really dynamic, ie also constantly changing names like ppp1.....xxxx the function will keep creating new IfDesc ad infinitum and never do any cleaning, which will cause memory leaks. It's not good habit to keep references to stuff that has really disappeared.
Third, it is called on every event loop iteration, so effectively every event. Rebuilding the interfaces takes considerable time compared to handling a request. So this may lead to unwanted delays, if on a larger network there are many requests to be handled.

So instead of not working correctly, it is now just not working. It will be when the rest of my changes come through.